### PR TITLE
Gemma3 Multimodal optimization

### PR DESCRIPTION
**Pull request by jiminha**
This is to optimize gemma3 multimodal memory/performance. 
 - bucket vision tower based on batch bucket to reduce recompile overhead 
 - modify merge_multimodal to use torch.where instead of masked_scatter for performance issue 
 - add warmup multimodal bucket to precompile vision tower 
 - port PT_HPU_SDPA_QKV_SLICE_MODE_FWD feature from vllm-fork v0 : this is necessary to reduce the memory for the longer sequence length. 
 - add 01,02 prefix for the general plugin so the order of initialization of plugins is guaranteed(ops followed by model)

### CODE REVIEW

**Comment by jiminha** [comment]
@xuechendi  Could you review this? This includes model file and some utils changes which are necessary for gemma3 model optimization. 
@adobrzyn  Could you review this? This includes multimodal warmup, added vision bucket into bucketing. Also ported PT_HPU_SDPA_QKV_SLICE_MODE_FWD feature from V0.

File: vllm_gaudi/v1/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -345,12 +345,29 @@ def __init__(self, model, vllm_config):
         self.is_mm_optimized = is_mm_optimized(self.model)
         self.sliding_window = vllm_config.model_config.get_sliding_window()
         self.interleaved_sliding_window = is_interleaved(vllm_config.model_config.hf_text_config)
+        if self.interleaved_sliding_window:
+            self.use_window_sdpa = os.getenv("PT_HPU_SDPA_QKV_SLICE_MODE_FWD", "false").strip().lower() in ("1", "true")
+            self.slice_size = int(os.getenv("PT_HPU_SDPA_BC_FACTOR", "1024"))
+            self.slice_thld = int(os.environ.get('VLLM_FUSEDSDPA_SLIDE_THLD', '8192'))
```
Comment (xuechendi) [inline vllm_gaudi/v1/worker/hpu_model_runner.py]: can we set default to extensions/features.py?

**Review by xuechendi — COMMENTED** [review]
PR overrall looks good to me, modeling changes and hpu_model_runner forward changes have been review in vllm-fork before. 
@adobrzyn @kzawora-intel , please check if bucketing changes looks good to you .

File: vllm_gaudi/extension/bucketing/common.py [hunk 2]
```diff
@@ -120,6 +132,13 @@ def generate_prompt_buckets(self):
                                                    self.max_num_seqs, self.max_num_prefill_seqs,
                                                    self.max_num_batched_tokens, self.block_size, self.num_hpu_blocks)
             self.log_generate_info(True)
+            if self.use_sliding_window:
+                self.prompt_buckets = [
+                    t for t in self.prompt_buckets
+                    if t[2] != 0 or (t[2] == 0 and (t[1] < self.slice_thld or
+                                                    (t[1] >= self.slice_thld and t[1] % self.slice_size == 0)))
+                ]
```
Comment (yafshar) [inline vllm_gaudi/extension/bucketing/common.py]:
```suggestion
                thld, size = self.slice_thld, self.slice_size
                self.prompt_buckets = [
                    t for t in self.prompt_buckets
                    if t[2] != 0 or t[1] < thld or t[1] % size == 0
                ]
```

minor, easier to read, identical outcome

**Review by xuechendi — APPROVED** [review]
